src/test/util/setup_common.h

```diff
@@ -44,6 +44,16 @@ std::ostream& operator<<(typename std::enable_if<std::is_enum<T>::value, std::os
 
 static constexpr CAmount CENT{1000000};
 
+/** Global network setup.
+ * On Windows, we globally terminate the use of the Winsock DLL when
+ * destructing the static instance_of_cnetcleanup. Consequently,
```

Comment:
> Why would `destructing the static instance_of_cnetcleanup` happen, and why does it matter?

https://github.com/bitcoin/bitcoin/blob/31ce305d46a191621aaa78a73fa4bbc7e58d8ed0/src/net.cpp#L3284-L3297

It happens because static objects are destructed at the end of binary execution. It matters because the desctructor calls `WSACleanup()` on Windows. As it happens once only, the `WSAStartup() must be called once only as well.

Reply:
None.

Reply:
> This PR fixes:
> 
>     1. Excessive network initialization in all unit tests and in some benchmarks, which use `MakeNoLogFileContext`.

In other words, `test_bitcoin.exe` calls `WSAStartup()` >500 times, and calls `WSACleanup()` exactly 1 time. According to Microsoft docs, these numbers must be equal.

Reply:
> > Lack of network initialization in other benchmarks.
> 
> Are fuzz tests not affected? Wouldn't it be easier to put the setup code into `libtest_util`, to cover all tests?

I will look into that. However, we do not use fuzzing on Windows, right?

Reply:
> Wouldn't it be easier to put the setup code into `libtest_util`, to cover all tests?

Sorry, if misunderstand your question, but this PR adds a new structure to `test/util/setup_common.{h,cpp}`. It already is a part of the `libtest_util` library.

> I don't use Windows, so I don't know whether it is possible to fuzz on Windows, or if anyone does it.

@dergoegge Do you have any related insights?

Reply:
> > Sorry, if misunderstood your question, but this PR adds a new structure to `test/util/setup_common.{h,cpp}`. It already is a part of the `libtest_util` library.
> 
> I mean similar to how the shutdown implicitly happens if you link the `net` module, the startup can happen (implicitly) by linking `libtest_util`, which should already happen for all tests?

Thanks! [Implemented](https://github.com/bitcoin/bitcoin/pull/28486#issuecomment-1830488289).